extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala

```diff
@@ -264,3 +237,33 @@ private[kinesis] object KinesisTestUtils {
     }
   }
 }
+
+/** A wrapper interface that will allow us to consolidate the code for synthetic data generation. */
+private[kinesis] trait KinesisProducer {
+  /** Sends the data to Kinesis possibly with aggregation if KPL is used. */
+  def putRecord(streamName: String, num: Int): Unit
+
+  /** Flush all data in the buffer and return the metadata for everything that has been sent. */
+  def flush(): Map[String, Seq[(Int, String)]]
```

Comment:
Why do you need this? Neither implementation seems to be doing any flushing. They only return the map. So why not simplify to one function that sends a bunch of data and returns the map.